File: plugin/src/main/java/com/stylegenerator/plugin/StyleGeneratorPluginTask.java
package com.stylegenerator.plugin;

import com.stylegenerator.plugin.util.ResourceReader;

import org.gradle.api.DefaultTask;
import org.gradle.api.tasks.TaskAction;

import java.io.File;

/**
 * Plugin task class.
 */
public class StyleGeneratorPluginTask extends DefaultTask {

    @TaskAction
    public void createFiles() throws Exception {

        final StyleGeneratorPluginExtension extension = getProject().getExtensions()
                .findByType(StyleGeneratorPluginExtension.class);

        final File outputDirectory = getProject().file("src/main/res/values");

        try {
            new StyleGenerator(
                    outputDirectory,
                    ResourceReader.getFontNames(),
                    ResourceReader.getColorNames(),
                    extension.parentStyle,
                    extension.minTextSize,
                    extension.maxTextSize
            ).generateResources();
        } catch (Exception e) {
            e.printStackTrace();
            throw e;
        }
    }
}

File: plugin/src/main/java/com/stylegenerator/plugin/StyleGeneratorPluginTask.kt
package com.stylegenerator.plugin

import com.stylegenerator.plugin.util.ResourceReader
import org.gradle.api.DefaultTask
import org.gradle.api.tasks.TaskAction

/**
 * Plugin task class.
 */
open class StyleGeneratorPluginTask : DefaultTask() {

    @TaskAction
    fun createFiles() {

        val extension = project.extensions
                .findByType(StyleGeneratorPluginExtension::class.java)
                ?: StyleGeneratorPluginExtension.DEFAULT

        val projectPath = project.projectDir.path

        try {
            StyleGenerator(
                    projectPath = projectPath,
                    fontNames = ResourceReader.getFontNames(projectPath),
                    colorNames = ResourceReader.getColorNames(projectPath),
                    parentStyle = extension.parentStyle,
                    generateUnusedStyles = extension.generateUnusedStyles,
                    minTextSize = extension.minTextSize,
                    maxTextSize = extension.maxTextSize
            ).generateResources()
        } catch (e: Exception) {
            e.printStackTrace()
            throw e
        }
    }
}
